src/js/game.js

```diff
@@ -0,0 +1,185 @@
+import { $, addEvent } from './utils.js';
+import { MESSAGE } from './constants.js';
+import Car from './car.js';
+
+export default class RacingGame {
+  constructor(container) {
+    this.container = container;
+    this.cars = [];
+    this.gameCount = 0;
+
+    this.init();
+    this.bindEvent();
+  }
+
+  init() {
+    this.carnameContainer = $({
+      target: this.container,
+      selector: '.carname-container',
+    });
+    this.carnameInput = $({
+      target: this.carnameContainer,
+      selector: 'input',
+    });
+    this.carnameButton = $({
+      target: this.carnameContainer,
+      selector: 'button',
+    });
+
+    this.gamecountContainer = $({
+      target: this.container,
+      selector: '.gamecount-container',
+    });
+    this.gamecountInput = $({
+      target: this.gamecountContainer,
+      selector: 'input',
+    });
+    this.gamecountButton = $({
+      target: this.gamecountContainer,
+      selector: 'button',
+    });
+
+    this.trackContainer = $({
+      target: this.container,
+      selector: '.track-container',
+    });
+    this.tracks = $({ target: this.container, selector: '.tracks' });
+
+    this.resultContainer = $({
+      target: this.container,
+      selector: '.result-container',
+    });
+    this.resultWinner = $({
+      target: this.resultContainer,
+      selector: '.winner',
+    });
+    this.restartButton = $({
+      target: this.resultContainer,
+      selector: '.restart',
+    });
+  }
+
+  bindEvent() {
+    addEvent({
+      el: this.carnameButton,
+      type: 'click',
+      callback: this.registerCars.bind(this),
+    });
+
+    addEvent({
+      el: this.gamecountButton,
+      type: 'click',
+      callback: this.registerCount.bind(this),
+    });
+
+    addEvent({
+      el: this.restartButton,
+      type: 'click',
+      callback: this.restartGame.bind(this),
+    });
+  }
+
+  registerCars = () => {
+    const carNames = this.carnameInput.value.split(',');
+    const validCarNames =
+      carNames &&
+      carNames.every(car => car.trim().length > 0 && car.trim().length <= 5);
+    if (!validCarNames) {
```

Comment:
괜히 car.trim 연산과 length 구하는 연산을 두번 하는 것 같은데, 아예 변수에 할당해놓는게 어떨까요  ?? 
```suggestion
    const validCarNames =
      carNames &&
      carNames.every(car =>{
      const length  = car.trim().length;
      return length > 0 && length <=5 
      });
    if (!validCarNames) {
```